apps/api/src/app/environments/usecases/update-widget-settings/update-widget-settings.usecase.ts

```diff
@@ -0,0 +1,37 @@
+import { Injectable } from '@nestjs/common';
+import { EnvironmentRepository } from '@novu/dal';
+import { UpdateWidgetSettingsCommand } from './update-widget-settings.command';
+import { ApiException } from '../../../shared/exceptions/api.exception';
+import { IWidgetSettings } from '@novu/dal';
+
+@Injectable()
+export class UpdateWidgetSettings {
+  constructor(private environmentRepository: EnvironmentRepository) {}
+
+  async execute(command: UpdateWidgetSettingsCommand) {
+    const environment = await this.environmentRepository.findById(command.environmentId);
+
+    if (!environment) {
+      throw new ApiException(`Environment id: ${command.environmentId} not found`);
+    }
+
+    const updateWidgetSettings: Partial<IWidgetSettings> = {};
+
+    if (command.widget.notificationCenterEncryption != null) {
+      updateWidgetSettings.notificationCenterEncryption = command.widget.notificationCenterEncryption;
+    }
+
+    await this.environmentRepository.update(
+      {
+        _organizationId: command.organizationId,
+        _id: command.environmentId,
+      },
+      { $set: { widget: updateWidgetSettings } }
```

Comment:
```suggestion
      { $set: updateWidgetSettings }
```

In this case in case a partial object will be generated in the if statements above, the widget object won't be accidently overridden in the future.

Reply:
Hey @scopsy , what i wanted to do here is to override only the widget property ,I am not sure I followed your suggestion I would love for further explanation if you don't mind or a quick example :)

Reply:
The main reason for square brackets is that you can have a "." using another syntax.
When using the dot inside the string, MongoDB updates only the internal key rather the replacing the entire widget object, nad only updates the specified key inside it. if it makes sense. I can jump on a quick call tomorrow if that was not clear enough :)